### PR TITLE
xmoto, fix build on x86

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
x86_gcc2 fails to build due to the libxml2 issue (https://github.com/haikuports/haikuports/issues/3890), _x86 builds but fails to start (fatal exeption: Unable to get xmDb version:)

File: games-sports/xmoto/xmoto-0.5.11.recipe [hunk 1]
```diff
@@ -9,77 +9,83 @@ COPYRIGHT="2006-2014 Nicolas Adenis-Lamarre
...
-	lib:libpng$secondaryArchSuffix
-	lib:libsdl$secondaryArchSuffix
-	lib:libsdl_mixer$secondaryArchSuffix
-	lib:libsdl_net$secondaryArchSuffix
-	lib:libsdl_ttf$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libSDL_1.2$secondaryArchSuffix
+	lib:libSDL_mixer_1.2$secondaryArchSuffix
+	lib:libSDL_net_1.2$secondaryArchSuffix
+	lib:libSDL_ttf_2.0$secondaryArchSuffix
```
Comment (alaviss) [inline games-sports/xmoto/xmoto-0.5.11.recipe]:
```suggestion
	lib:libSDL_ttf_2.0$secondaryArchSuffix
```

Shouldn't this be 1.2 or removed? We can't mix different version of sdl like that...

Comment (pulkomandy) [reply games-sports/xmoto/xmoto-0.5.11.recipe]: SDL ttf up to 2.0.11 goes with SDL 1.2 (https://www.libsdl.org/projects/SDL_ttf/release-1.2.html)
Not sure what the so versions are...

Comment (alaviss) [reply games-sports/xmoto/xmoto-0.5.11.recipe]: The other lib would be libSDL2_ttf it seems, so this isn't much of a problem.

Comment (Begasus) [reply games-sports/xmoto/xmoto-0.5.11.recipe]: Issue for this is at https://github.com/haikuports/haikuports/issues/3878

**Comment by korli** [comment]
maybe switch back to gcc2?

**Comment by Begasus** [comment]
I'll check in later once the libxml2 is in the depot :)

**Comment by Begasus** [comment]
@korli problem with libxml2 is fixed, thanks for that! But disabled the build for gcc2 due to too many parse errors, build for _x86 is ok but still can't launch it (see comment above), marked broken and disabled for now

**Comment by korli** [comment]
Ok thanks. I thought the game was old and originally with gcc2.

**Comment by Begasus** [comment]
It is old, no new changes since first merge for this, that's why it's strange it doesn't build anymore, maybe something changed in our gcc2 that broke it?

**Comment by Begasus** [comment]
Got past the parse errors but at the end ran into:
```
boot/home/WIP/xmoto-0.5.11/src/xmscene/ChipmunkWorld.cpp:178: Internal compiler error, output_operand_lossage `invalid expression as operand'
```
Guess that's the end?

**Comment by korli** [comment]
you could try https://github.com/xmoto/xmoto and switch to x86.

**Comment by Begasus** [comment]
Not really a success with our libode (I think)
xmoto-github

**Comment by Begasus** [comment]
@korli bumped to the github latest commit there, it builds ok in Terminal, not tested with haikuporter yet, but wanted to share the work so far already

**Comment by korli** [comment]
What's defined in ode/precision.h?

**Comment by Begasus** [comment]
@korli check pkg-config-x86 ode --cflags it gives you -dSINGLE

**Comment by Begasus** [comment]
Latest push needs "cd build" in INSTALL before "make install", (build is ok as it seems) but that still fails on 
```
file INSTALL cannot find
"sources/xmoto-$srcGitRev/build/bin/xmoto.bin"
```
That was the problem before also when trying to build from Terminal setting the prefix ...

**Comment by Begasus** [comment]
Latest push needs "cd build" in INSTALL before "make install", (build is ok as it seems) but that still fails on 
```
file INSTALL cannot find
"sources/xmoto-$srcGitRev/build/bin/xmoto.bin"
```
That was the problem before also when trying to build from Terminal setting the prefix ...

> What's defined in ode/precision.h?

Seems this is created at configure time, the next comes from ode 0.16
```
#ifndef _ODE_PRECISION_H_
#define _ODE_PRECISION_H_

/* Define dSINGLE for single precision, dDOUBLE for double precision,
* but never both!
*/

#if defined(dIDESINGLE)
#define dSINGLE
#elif defined(dIDEDOUBLE)
#define dDOUBLE
#else
#define dSINGLE
#endif

#endif
```

**Comment by Begasus** [comment]
If no one objects, I'm closing this one, even with the patches I can't get xmoto to start

**Pull request merged**